frontend/model/contracts/group.js

```diff
@@ -1147,6 +1147,31 @@ sbp('chelonia/defineContract', {
         if (!state.generalChatRoomId) {
           Vue.set(state, 'generalChatRoomId', data.chatRoomID)
         }
+      },
+      async sideEffect ({ data, meta, contractID }, { state: Rstate }) {
+        const rootState = sbp('state/vuex/state')
+        const contracts = rootState.contracts || {}
+        const { identityContractID } = rootState.loggedIn
+        const userState = rootState[identityContractID]
+        const state = rootState[contractID]
+
+        if (rootState[data.chatRoomID]?._volatile) {
+          return
+        }
+
+        await sbp('chelonia/out/keyRequest', {
+          originatingContractID: identityContractID,
+          originatingContractName: contracts[identityContractID].type,
+          contractID: data.chatRoomID,
+          contractName: 'gi.contracts/chatroom',
+          signingKey: state._volatile?.keys?.[(((Object.values(Object(state._vm?.authorizedKeys)): any): GIKey[]).find((k) => k?.meta?.type === 'csk')?.id: ?string)],
+          innerSigningKeyId: ((Object.values(userState._vm.authorizedKeys): any): GIKey[]).find((k) => k.meta?.type === 'csk')?.id,
+          encryptionKeyId: ((Object.values(userState._vm.authorizedKeys): any): GIKey[]).find((k) => k.meta?.type === 'cek')?.id,
+          hooks: {
+            prepublish: null,
+            postpublish: null
+          }
+        })
```

Comment:
Could you add a comment explaining the specific scenario that's being handled here?

Reply:
I'll do; the specific scenario here is joining chatrooms (should be the general channel in this case) . This is required because joining a chatroom is a two-step process (i.e., first a group is joined, we get the list of chatrooms and then request to join) due to them being encrypted. In the future, this could be optimised so that multiple contracts can be 'joined' (for want of a better word) in a single operation

Reply:
Update: we agreed that this code should be removed so that a different method of obtaining the chatroom key can be implemented instead (by pre-sharing the key in the group contract using OP_KEYSHARE)